sql/20201027-routed.sql

```diff
@@ -0,0 +1,38 @@
+ALTER TABLE threads
+ADD COLUMN routed boolean;
+
+
+--- populate routed threads
+WITH threads_with_rank AS (
+    SELECT *, row_number() OVER (PARTITION BY user_id, is_demo ORDER BY updated_at DESC) AS rn
+    FROM threads
+)
+UPDATE threads
+SET routed='t'
+WHERE EXISTS (
+    SELECT 1 FROM threads_with_rank r
+    WHERE
+        r.slack_channel_id = threads.slack_channel_id
+        AND r.slack_parent_message_ts = threads.slack_parent_message_ts
+        AND r.user_id = threads.user_id
+        AND r.is_demo = threads.is_demo
+        AND rn > 1
```

Comment:
If a voter is routed from lobby to channel A to channel B, won't the thread for channel A have rn = 2 and the thread for channel B have rn = 3, but both will end up with routed = true even though channel A's thread is inactive?

maybe this is me still not understanding what routed is. regardless, can you please add more comments about what you're intending to happen in these SQL queries and what routed is supposed to represent, both here and in other files that deal with the routed logic? I want this to be easy to read and understand for future developers.

Reply:
I understand it now (missed that you had updated updated_at DESC). Can you add comments explaining that you’re marking all but the latest threads routed (or inactive, ideally).